### DISCUSSION TITLE
The transition

### DISCUSSION THREAD

**Discussion by Vivraan** [Godot-Languages-Support/godot-lang-support#61] [General]
Hey, Godot lovers!

It seems we need to begin the discussion of the transition documentation correctly now that Godot 4 is out and in full swing, with only a few applications restricted to Godot 3.

Authors, it might be of interest if we could get some volunteers to make a GitHub page or similar that we could host off of this repository (owing to its existing popularity) and update details for GDExtension-based language plugins and the Godot 3-related stuff.

We hosted similar discussions a long time ago in Issues, during which @ShalokShalom (the author of the original GitHub Gist) concurred to a similar idea. Of course, now that we have Godot 4, we should get down to the task of getting the new face published!

I was responsible for creating the present README file, and with loads of contributors filling in, it was a pleasure to keep updating the list. Unfortunately, I have become busy with life since I created this listing about 4 years ago. Godot has become something I seldom dabble with, meaning I have lost touch with the community effort. It would be wise if a new team took over active maintenance of the new page; I'm happy to chime in as a community member, regardless, and would like to hand over my Maintainer rights to a more active, dedicated, and perhaps bigger Godot-lover than myself.

This is not to say goodbye -- oh no, Godot yet thrives, and I, for one, would like to return to the effort at some point as an engine and plugin contributor -- my busy life has made me a better programmer. B).

Let's continue this discussion here.

Best,
Shivam/Vivraan.

EDITS:
1. @ShalokShalom and @winston-yallow are now officially maintainers/owners of the Godot Language Support GitHub org. I am still an owner to facilitate the transition to the website. Once the website development is fully underway and all maintainers (we may need of some more) can contribute meaningfully to the management of the project, I'll relieve myself of ownership.
2. Discussion about the website should be moved to #63.

**Comment by winston-yallow** [comment]
I would be open to work on transforming the current README into something that can be used as a GitHub page. Though I do not want to become sole maintainer as I have too many other things going on already.

**Reply by Vivraan** [reply]
That'd be awesome! We did try out a GitHub deployment some time ago, but it's been a while since.

**Reply by ShalokShalom** [reply]
If this homepage is done, I can also help with maintaining it.

**Comment by Vivraan** [comment]
Are we maintaining the same emoji-based style? It does work even now, but some manner of tagging might be useful if we want to add a "better" search functionality or similar.

**Reply by winston-yallow** [reply]
Good question. Also depends a bit how we keep the data in the future. It maybe makes sense to have the data in a yaml file or similar, and generate a static website based on it with a GitHub action.

This approach would be nice since it enables to easily change the representation without having to modify each individual entry. It would also allow to easily add filtering / sorting to the list.

**Reply by Vivraan** [reply]
Model-view approach is amenable! Having it at YAML is also smart so we don't need extra logic to render things!

**Reply by winston-yallow** [reply]
I won't have much time this month, but I'll probably try and make a PR next month for this.

If you feel comfortable giving me maintainer rights, I could also look into setting up an action to deploy it as a GitHub page

**Reply by ShalokShalom** [reply]
I would strongly suggest so.

**Reply by Vivraan** [reply]
Let me know when we can start working towards it. I'd be happy to port the contents to YAML if I get a scaffolding!

**Reply by Vivraan** [reply]
As for maintainer rights, I believe I must make an org for that, right? We'd transfer the repo to the org and I'd probably have the option to control rights org-wide. Not entirely sure since I use GitLab mostly. 😅

For this, is it okay if I make it a business rather than a personal org? We'll need a contact email, etc. if it's not personal.

**Reply by Croydon** [reply]
> For this, is it okay if I make it a business rather than a personal org? We'll need a contact email, etc. if it's not personal.

You probably meant this the other way around, right? It is a community effort, so it is a "personal org"

**Reply by ShalokShalom** [reply]
If we make this a proper static website, we could host this just as much on Codeberg

**Reply by Vivraan** [reply]
> > For this, is it okay if I make it a business rather than a personal org? We'll need a contact email, etc. if it's not personal.
> 
> You probably meant this the other way around, right? It is a community effort, so it is a "personal org"

Well, I don't know the specifics of that, unfortunately. Let's try something?

**Comment by Vivraan** [comment]
Here's the new org-based link to the repo:

https://github.com/Godot-Languages-Support/godot-lang-support/discussions/61

Once @winston-yallow and @ShalokShalom accept the invitations, I'll transfer ownership/admin roles to them.

**Reply by Vivraan** [reply]
@ShalokShalom just granted you ownership. You should be able to manage everything!

**Reply by Vivraan** [reply]
@winston-yallow ready when you are!

**Reply by winston-yallow** [reply]
I've accepted the invitation! Sorry for not replying earlier, this month is incredibly busy for me. But as I said, I will have time next month to create the static website :tada:

**Comment by Vivraan** [comment]
@winston-yallow @ShalokShalom should we create a new discussion thread for the website?

**Reply by ShalokShalom** [reply]
We can

**Reply by ShalokShalom** [reply]
As a note:

Projects die all the time, and we might better check on the different implementation regularly. 
Maybe there is a tool, that reports on how many commits a project does. 

We then can show with a badge/widget how active the project it. 
And maybe send as a warning, if it becomes stale. 

So we can rerank it quickly.

**Reply by Vivraan** [reply]
> We then can show with a badge/widget how active the project it. And maybe send as a warning, if it becomes stale.
> 
> So we can rerank it quickly.

I guess that's going to need GitHub API to check for activity. The easiest would be "last commit was [x time] ago".

**Reply by ShalokShalom** [reply]
Yeah, something like that sounds good.

**Reply by Vivraan** [reply]
https://github.com/Godot-Languages-Support/godot-lang-support/discussions/63

Let's proceed with discussions about the website here.
